plugins/gfonts.koplugin/main.lua

```diff
@@ -0,0 +1,283 @@
+        width = Screen:getWidth() - 15,
+        height = Screen:getHeight() - 15,
+        show_parent = menu_container,
+        onMenuHold = self.onMenuHold,
+        cface = Font:getFace("smallinfofont"),
+        _manager = self,
+    }
+    table.insert(menu_container, self.search_menu)
+    self.search_menu.close_callback = function()
+        UIManager:close(menu_container)
+    end
+    self.search_menu:switchItemTable(title, t)
+    UIManager:show(menu_container)
+end
+
+function GFonts:close()
+    self.search_dialog:onClose()
+    UIManager:close(self.search_dialog)
+    if self.lastsearch then
+        self:find(self.lastsearch)
+    end
+end
+
+function GFonts:find(option)
+    local catalog, catalog_name
+    if self.lastsearch == "featured" then
+        local recommended = featured:getFonts(self.fonts.list)
+        logger.info(recommended)
+        catalog = self:fontCatalog(recommended)
+        catalog_name = _("Featured fonts")
+    elseif self.lastsearch == "family" then
+        catalog = self:fontCatalog(self.fonts.list)
+        catalog_name = _("Fonts")
+    else
+        if self.lastsearch == "category" then
+            catalog_name = _("Categories")
+        elseif self.lastsearch == "language" then
+            catalog_name = _("Languages")
+        elseif self.lastsearch == "last" then
+            catalog_name = _("Last modified")
+        else
+            logger.info(self.lastsearch)
+            return
+        end
+        local results = fontsearch:sortBy(self.lastsearch, self.fonts.list)
+        catalog = self:optionsCatalog(results, self.lastsearch)
+    end
+    if catalog and catalog_name then
+        self:showResults(catalog, catalog_name)
+    end
+end
+
+function GFonts:onMenuHold(item)
+    if not item.info or item.info:len() <= 0 then return end
+    UIManager:show(InfoMessage:new{
+        text = item.info,
+    })
+end
+
+function GFonts:optionsCatalog(t, option)
+    local catalog = {}
+    for item, ocurrence in pairs(t) do
+        local entry = {}
+        local name
+        if option == "last" then
+            if item == "last_year" then
+                name = _("Last year")
+            elseif item == "last_month" then
+                name = _("Last month")
+            elseif item == "last_week" then
+                name = _("Last week")
+            else
+                name = _("Older")
+            end
+        else
+            name = item
+        end
+        entry.text = string.format("%s (%d)", name, ocurrence)
+        entry.callback = function()
+            local fonts = self:fontCatalog(fontsearch:fontsBy(option, item, self.fonts.list))
+            self:showResults(fonts, _("Fonts"))
+        end
+        table.insert(catalog, entry)
+    end
+    return catalog
+end
+
+
+function GFonts:downloadFont(family)
+    logger.info("Prompt download " .. family)
+    UIManager:show(ConfirmBox:new{
+        text = T(_("Do you want to download %1"), family),
```

Comment:
Maybe short and to the point? Either way I'd probably add that question mark.
```suggestion
        text = T(_("Download %1?"), family),
```